### DISCUSSION TITLE
questions about how to model SCR data in my case

### DISCUSSION THREAD

**Discussion by TeaforTwo1011** [bachlab/PsPM#444] [Q&A] [answered]
Hi Dominik,

I have a study design like this: 40 participants, each has 54 trials, and each trial look like this:

CS -> Anticipatory SCR -> US (Shock/ Shock omission)

What I am interested is modeling anticipatory SCR in a time series way, and the CS is just static visual input, so I thought GLM with 54 conditions would be suitable, and my event markers is on time points that just followed by CS.

However, my reconstructed response looks like this, which is clearly problematic:
<img width="960" alt="1672919438293" src="https://user-images.githubusercontent.com/92511991/210774034-d541932d-5f6a-4792-b634-f85fcb3bc8f7.png">

I have checked my timing and there is no problem with it, the only reason that I can think of is because there are huge uphills after US onset. All those peaks that you can see from the continuous data appears when there are shocks following, which interfered with my estimated anticipatory SCR because their amplitudes are not high compared to what shocks led to:
<img width="960" alt="1672919555411" src="https://user-images.githubusercontent.com/92511991/210774282-901a0689-39fb-408f-9441-cee1c0356c02.png">

Do you have any idea about how to improve this analysis? I can also create event markers for the US onsets, but I am interested is what is happened between CS onset and US onset, which is a interval around 5s.

Thank you very much, 
Yufei

**Comment by teddphil** [comment]
Hi @TeaforTwo1011 

Thank you for proposing this question.
Just to confirm with you about the question. I noticed this question was talking about an issue of importing SCR file into PsPM, and now it is changed to improving SCR analysis. May I know if the issue of file importing has been resolved, and it is only of interest to improve the analysis?

Best wishes 
Teddy

**Reply by TeaforTwo1011** [reply]
Hi Teddy,

yes I have solved the question of importing text file by looking into the example dataset from the tutorial that available online :) And now my remaining question is about how to improve the analysis. Let me know if there is anything unclear and thanks for your help!

**Comment by dominikbach** [comment, marked as answer]
From what I can see, each trial consists of two events (CS, US) but you are only modelling one of them (presumably the trial onset, i.e. CS)? This could explain your finding of a negative reconstructed CS response. 

See also https://github.com/bachlab/PsPM/discussions/440

Dominik

**Reply by TeaforTwo1011** [reply]
Thanks Dominik!

I also read the answer under #440, but I still have a question:

Does that mean I need to add another marker for the US (also 54 trials), and I will end up with a GLM with 108 conditions, but only look at the parameters in trials that I am interested in?

Or, is non-linear model better than GLM in this case ?

Thank you very much!

**Comment by dominikbach** [comment]
1. Yes, we would recommend also modelling the US. Unless you want to analyse trial-by-trial response, it would be better to set up a condition-wise model, i.e. put each trial of the same type into the same condition. 
2. Non-linear models for fear conditioning yield more precise estimates in many situations.
